core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentUpdateStatusManager.java

```diff
@@ -450,8 +450,7 @@ public boolean accept(CarbonFile pathName) {
             String fileName = pathName.getName();
             if (fileName.endsWith(CarbonCommonConstants.DELETE_DELTA_FILE_EXT)
                 && pathName.getSize() > 0) {
-              String firstPart = fileName.substring(0, fileName.indexOf('.'));
-              String blkName = firstPart.substring(0, firstPart.lastIndexOf("-"));
+              String blkName = fileName.substring(0, fileName.lastIndexOf("-"));
```

Comment:
Can you please add a testcase to check num of files after update/delete in a compacted segment

Reply:
@Indhumathi27 added test case